code/modules/mechanics/MechanicMadness.dm

```diff
@@ -4129,6 +4129,240 @@ ADMIN_INTERACT_PROCS(/obj/item/mechanics/trigger/button, proc/press)
 		// so you can tell if scrimblo made a cool scene and then dogshit2000 put obscenities on top or whatever.
 		logTheThing(LOG_STATION, null, "draws on [src]: [log_loc(src)]: canvas{\ref[src], [x], [y], [dot_color], [x2], [y2]}")
 
+/// exists just to allow the grabber to attack
+/mob/living/grabber_goblin
+	name = "\improper Grabber Component"
+	desc = "you shouldn't see this!"
+	canmove = FALSE
+	invisibility = INVIS_ALWAYS
+	use_stamina = FALSE
+	canbegrabbed = FALSE
+	anchored = ANCHORED_ALWAYS
+	nodamage = TRUE
+
+	var/obj/item/mechanics/grabber/home_grabber
+
+	New(atom/loc)
+		. = ..()
+		if (loc)
+			name = loc.name
+			real_name = loc.name
+			if (istype(loc,/obj/item/mechanics/grabber))
+				home_grabber = loc
+
+	// the goblin in the grabber arm dying is probably bad
+	death()
+		if (home_grabber == src.loc)
+			return
+		. = ..()
+
+	ex_act()
+		return
+
+	disposing()
+		if(istype(home_grabber))
+			home_grabber.grabber_goblin = null
+			home_grabber = null
+		. = ..()
+
+
+/obj/item/mechanics/grabber
+	// picks up items, can spin, and can smack something with it
+	name = "\improper Grabber Component"
+	desc = "Usable to grab items and create a mini assembly line"
+	icon_state = "grabber"
+	cabinet_banned = TRUE
+	plane = PLANE_DEFAULT
+	density = TRUE
```

Comment:
A dense item you can pick up seems dangerous, I can imagine people using this to block stuff and cause jank. Either make it only dense when anchored or just not dense at all since I don't think any other mechcomp component is?